### ISSUE TITLE
Run optimize-vscode in Travis CI

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#7735]
This error seems to occur pretty frequently, we should have Travis CI run optimize-vscode so it's caught more that once a day.

```
[11:19:38] Starting 'optimize-vscode'...
evalmachine.<anonymous>:935
            throw new Error('Could not correlate define call site for needle ' + needle);
            ^

Error: Could not correlate define call site for needle node|vs/code/electron-main/|ansi-regex
    at Function.ModuleManager._findRelevantLocationInStack (evalmachine.<anonymous>:935:19)
    at evalmachine.<anonymous>:949:78
    at Array.map (native)
    at ModuleManager.getBuildInfo (evalmachine.<anonymous>:942:47)
    at Function.RequireFunc.getBuildInfo (evalmachine.<anonymous>:1837:34)
    at Object.<anonymous> (/home/daniel/dev/Microsoft/vscode/build/lib/bundle.js:35:30)
    at Module._invokeFactory (evalmachine.<anonymous>:764:48)
    at Module._complete (evalmachine.<anonymous>:776:34)
    at Module.resolveDependency (evalmachine.<anonymous>:831:22)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1190:39)
```

**Comment by headmelted** [comment]
Any idea what was actually causing this?

I'm now getting it consistently when I target armhf.

**Comment by Tyriar** [comment]
@joaomoreno might remember?

**Comment by joaomoreno** [comment]
@headmelted Can you show us the entire output you see?

**Comment by headmelted** [comment]
$ gulp optimize-vscode;
[10:39:53] Using gulpfile /builds/headmelted/code-builds/gulpfile.js
[10:39:53] Starting 'clean-optimized-vscode'...
[10:39:53] Starting 'clean-client-build'...
[10:39:53] Starting 'clean-extension:configuration-editing'...
[10:39:53] Starting 'clean-extension:css-client'...
[10:39:53] Starting 'clean-extension:css-server'...
[10:39:53] Starting 'clean-extension:extension-editing'...
[10:39:53] Starting 'clean-extension:git'...
[10:39:53] Starting 'clean-extension:gulp'...
[10:39:53] Starting 'clean-extension:html-client'...
[10:39:53] Starting 'clean-extension:html-server'...
[10:39:53] Starting 'clean-extension:javascript'...
[10:39:53] Starting 'clean-extension:json-client'...
[10:39:53] Starting 'clean-extension:json-server'...
[10:39:53] Starting 'clean-extension:markdown'...
[10:39:53] Starting 'clean-extension:php'...
[10:39:53] Starting 'clean-extension:python'...
[10:39:53] Starting 'clean-extension:ruby'...
[10:39:53] Starting 'clean-extension:typescript-test-colorize-fixtures'...
[10:39:53] Starting 'clean-extension:typescript'...
[10:39:53] Starting 'clean-extension:vscode-api-tests'...
[10:39:53] Starting 'clean-extension:vscode-colorize-tests'...
[10:39:53] Finished 'clean-optimized-vscode' after 49 ms
[10:39:53] Finished 'clean-client-build' after 43 ms
[10:39:53] Starting 'compile-client-build'...
[10:39:54] Starting compilation...
[10:39:54] Finished 'clean-extension:typescript-test-colorize-fixtures' after 201 ms
[10:39:54] Starting 'compile-extension-build:typescript-test-colorize-fixtures'...
[10:39:54] Finished 'compile-extension-build:typescript-test-colorize-fixtures' after 130 ms
[10:39:54] Finished 'clean-extension:css-client' after 553 ms
[10:39:54] Starting 'compile-extension-build:css-client'...
[10:39:54] Finished 'clean-extension:css-server' after 570 ms
[10:39:54] Starting 'compile-extension-build:css-server'...
[10:39:54] Finished 'clean-extension:configuration-editing' after 590 ms
[10:39:54] Starting 'compile-extension-build:configuration-editing'...
[10:39:54] Finished 'clean-extension:extension-editing' after 601 ms
[10:39:54] Starting 'compile-extension-build:extension-editing'...
[10:39:54] Finished 'clean-extension:git' after 647 ms
[10:39:54] Starting 'compile-extension-build:git'...
[10:39:54] Finished 'clean-extension:gulp' after 652 ms
[10:39:54] Starting 'compile-extension-build:gulp'...
[10:39:54] Finished 'clean-extension:html-client' after 660 ms
[10:39:54] Starting 'compile-extension-build:html-client'...
[10:39:54] Finished 'clean-extension:json-client' after 687 ms
[10:39:54] Starting 'compile-extension-build:json-client'...
[10:39:54] Finished 'clean-extension:markdown' after 707 ms
[10:39:54] Starting 'compile-extension-build:markdown'...
[10:39:54] Finished 'clean-extension:python' after 720 ms
[10:39:54] Starting 'compile-extension-build:python'...
[10:39:54] Finished 'clean-extension:ruby' after 753 ms
[10:39:54] Starting 'compile-extension-build:ruby'...
[10:39:54] Finished 'clean-extension:vscode-api-tests' after 804 ms
[10:39:54] Starting 'compile-extension-build:vscode-api-tests'...
[10:39:54] Finished 'clean-extension:vscode-colorize-tests' after 808 ms
[10:39:54] Starting 'compile-extension-build:vscode-colorize-tests'...
[10:39:55] Finished 'clean-extension:javascript' after 1.71 s
[10:39:55] Starting 'compile-extension-build:javascript'...
[10:39:55] Finished 'clean-extension:json-server' after 1.72 s
[10:39:55] Starting 'compile-extension-build:json-server'...
[10:39:55] Finished 'clean-extension:typescript' after 1.72 s
[10:39:55] Starting 'compile-extension-build:typescript'...
[10:39:56] Finished 'clean-extension:php' after 2.39 s
[10:39:56] Starting 'compile-extension-build:php'...
[10:40:06] Finished 'clean-extension:html-server' after 12 s
[10:40:06] Starting 'compile-extension-build:html-server'...
[10:40:06] [i18n] Message file chs/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file cht/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file jpn/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file kor/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file deu/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file fra/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file esn/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file rus/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:06] [i18n] Message file ita/extensions/gulp/out/main.i18n.json not found. Missing messages: 1
[10:40:21] Finished 'compile-extension-build:python' after 27 s
[10:40:21] Finished 'compile-extension-build:ruby' after 27 s
[10:40:23] Finished 'compile-extension-build:css-server' after 29 s
[10:40:28] Finished 'compile-extension-build:vscode-colorize-tests' after 34 s
[10:40:28] [i18n] Message file chs/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file cht/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file jpn/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file kor/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file deu/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file fra/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file esn/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file rus/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:28] [i18n] Message file ita/extensions/markdown/out/security.i18n.json not found. Missing messages: 5
[10:40:35] Finished 'compile-extension-build:json-server' after 40 s
[10:40:35] Finished 'compile-extension-build:gulp' after 41 s
[10:40:40] Finished 'compile-extension-build:css-client' after 46 s
[10:40:40] Finished 'compile-extension-build:configuration-editing' after 46 s
[10:40:40] Finished 'compile-extension-build:extension-editing' after 46 s
[10:40:40] Finished 'compile-extension-build:json-client' after 46 s
[10:40:40] Finished 'compile-extension-build:html-client' after 46 s
[10:40:43] Finished 'compile-extension-build:vscode-api-tests' after 49 s
[10:40:43] [i18n] Message file chs/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file cht/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file jpn/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file kor/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file deu/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file fra/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file esn/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file rus/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:43] [i18n] Message file ita/extensions/git/out/scmProvider.i18n.json not found. Missing messages: 1
[10:40:48] Finished 'compile-extension-build:javascript' after 53 s
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:48] [i18n] No localized message found for key typesInstallerInitializationFailed
[10:40:49] Finished 'compile-extension-build:php' after 53 s
[10:40:49] Finished 'compile-extension-build:markdown' after 54 s
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] [i18n] No localized message found for key open
[10:40:49] Finished 'compile-extension-build:html-server' after 43 s
[10:40:49] [i18n] No localized message found for key confirm discard multiple
[10:40:50] Finished 'compile-extension-build:git' after 56 s
[10:40:50] [i18n] No localized message found for key openTsServerLog.openFileFailedFailed
[10:40:50] [i18n] No localized message found for key openTsServerLog.openFileFailedFailed
[10:40:50] Finished 'compile-extension-build:typescript' after 55 s
[10:40:50] Starting 'compile-extensions-build'...
[10:40:50] Finished 'compile-extensions-build' after 73 μs
[10:44:52] [monaco.d.ts] Starting monaco.d.ts generation
[10:44:56] [monaco.d.ts] Finished monaco.d.ts generation
[10:44:56] [memory watch dog] memory usage after compilation finished: 911.81 MB
[10:44:56] [memory watch dog] Reloading typescript node module...
[10:44:56] [memory watch dog] typescript node module reloaded.
[10:44:56] Finished compilation with 0 errors after 302713 ms
[10:44:56] Finished 'compile-client-build' after 5.03 min
[10:44:56] Starting 'compile-build'...
[10:44:56] Finished 'compile-build' after 50 μs
[10:44:56] Starting 'optimize-vscode'...
evalmachine.<anonymous>:955
            throw new Error('Could not correlate define call site for needle ' + needle);
            ^

Error: Could not correlate define call site for needle node|vscode-ripgrep
    at Function.ModuleManager._findRelevantLocationInStack (evalmachine.<anonymous>:955:19)
    at ModuleManager.getBuildInfo (evalmachine.<anonymous>:973:80)
    at Function.RequireFunc.getBuildInfo (evalmachine.<anonymous>:1517:34)
    at Object.<anonymous> (/builds/headmelted/code-builds/build/lib/bundle.js:53:30)
    at Function.Module._invokeFactory (evalmachine.<anonymous>:815:41)
    at Module.complete (evalmachine.<anonymous>:825:36)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1412:20)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
    at ModuleManager._onModuleComplete (evalmachine.<anonymous>:1424:30)
ERROR: Job failed: exit code 1

**Comment by headmelted** [comment]
Thanks for coming back on this.

I can get the verbose logs for the earlier steps if needs be, but will need to re-compile to get the file dumped.

**Comment by joaomoreno** [comment]
cc @roblourens Strange that `vscode-ripgrep` breaks this.

**Comment by roblourens** [comment]
What is optimize-vscode trying to do exactly? vscode-ripgrep's postinstall task may have failed for armhf, not sure.

**Comment by headmelted** [comment]
Sorry guys, I thought I'd closed out the vscode-ripgrep issue before
raising here, but I was mistaken.

The reason for raising here is that I don't believe this second build error
is related to vscode-ripgrep (now getting past that point in the build
script, so will get the other ticket closed).

If nothing comes to mind then I'll have to assume it's due to something I'm
doing that's no longer working with the latest commits, and have another
look at my scripts.

Thanks for trying to help me out on this though! :-)

**Comment by joaomoreno** [comment]
Does `vscode-ripgrep` fail to install on your machine, @headmelted ?

**Comment by headmelted** [comment]
It's been working since I changed my armv7l references to "arm" to match
the script in vscode-ripgrep (as best as I can tell at least. I've since
been getting the above message so still no builds since it started
happening for me to test).

**Comment by roblourens** [comment]
I don't quite understand, so vscode-ripgrep install isn't failing, but you're still getting that message?

If you had to make a change for vscode-ripgrep, I'd appreciate if you open an issue on that repo with details. I haven't really tested it on arm.

**Comment by headmelted** [comment]
Sorry, I've caused some confusion here.

I raised a query on the @Tyriar vscode-ripgrep ticket here:

https://github.com/roblourens/vscode-ripgrep/issues/1

My part of _**that**_ ticket can be safely ignored, I've explained in the ticket why - but basically I changed my scripts from passing "armv7l" to passing "arm".

This new issue does name vscode-ripgrep, but I don't believe this new problem is due to vscode-ripgrep, seeing as how you folks aren't experiencing this issue.

I am scratching my head a little with this one though.

I may need to just dig through my code and vscode-ripgrep just to see if there's something else I'm missing.  Thanks for taking the time in any case guys, it's much appreciated.
